### PR TITLE
feat(python): lazyframe __repr__ returns schema

### LINKED ISSUE #9100: Should py-polars LazyFrame's __repr__ return its schema?
### Problem description

I wish that the polars LazyFrame's `__repr__` would return its schema.

When I'm developing in debug mode, I frequently call a dataframe's `__repr__` method to see its current state, do an operation on it, then call `__repr__` again to see the change. This works excellent with Pandas' dataframe, Polars' dataframe etc.

However, for the `LazyFrame`, the `__repr__` currently tells me that it's an instance of the `LazyFrame` class and its position in memory. This doesn't help me develop my application at all, and is inconsistent with the `DataFrame`.

IMO, the closest we can get is to show the schema.

What do you guys think? I've submitted a PR at #9101.

Also, thanks a tonne for an **excellent** library! It's made my job a lot more fun 🥳

### PR DESCRIPTION
**Pull request by MartinBernstorff**
See the accompanying issue for discussion, if merged, closes #9100.

Was a one-liner, so thought I might as well 👍

### CODE REVIEW

File: py-polars/polars/lazyframe/frame.py [hunk 1]
```diff
@@ -670,9 +670,9 @@ def __str__(self) -> str:
 {self.explain(optimized=False)}\
 """
 
-    def __repr__(self) -> str:
+    def __repr__(self) -> Mapping[str, Union[DataTypeClass, DataType]]:
```
Comment (MarcoGorelli) [inline py-polars/polars/lazyframe/frame.py:673]: the return type needs to be a `str`, not `Mapping`, could you update that please? might need to check if there's a test which needs updating too (and if there isn't one, to write one)

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
at a first glance, this seems like a nice change, nice one for suggesting it! curious if there's objections

**Comment by alexander-beedie** [comment]
Not convinced about this; the schema is not the frame, and is readily available from `.schema()` if that's what you want to see. 

However, I do agree with the sentiment behind it; we could clearly do something more informative than what we currently have ;)

**Comment by MartinBernstorff** [comment]
Happy to get this started! I've posted a comment in #9100, perhaps we should keep the discussion there for now 👍

**Pull request closed without merging**